cmd/shimdiag/list.go

```diff
@@ -1,24 +1,57 @@
 package main
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/Microsoft/hcsshim/internal/appargs"
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
 	"github.com/urfave/cli"
 )
 
 var listCommand = cli.Command{
 	Name:      "list",
 	Usage:     "Lists running shims",
 	ArgsUsage: " ",
-	Before:    appargs.Validate(),
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "pids",
+			Usage: "Shows the process IDs of each shim",
+		},
+	},
+	Before: appargs.Validate(),
 	Action: func(ctx *cli.Context) error {
+		pids := ctx.Bool("pids")
 		shims, err := findShims("")
 		if err != nil {
 			return err
 		}
+
+		getPid := func(shimStr string) (int32, error) {
+			shim, err := getShim(shimStr)
+			if err != nil {
+				return 0, err
+			}
+			defer shim.Close()
+
+			svc := shimdiag.NewShimDiagClient(shim)
+			resp, err := svc.DiagPid(context.Background(), &shimdiag.PidRequest{})
+			if err != nil {
+				return 0, err
+			}
+			return resp.Pid, nil
+		}
+
 		for _, shim := range shims {
-			fmt.Println(shim)
+			output := shim
+			if pids {
+				pid, err := getPid(shim)
+				if err != nil {
+					return err
+				}
+				output = fmt.Sprintf("%s ==> %d", output, pid)
```

Comment:
I would prefer we just output results in a table form like: 

```
Shims                          PID
k8s.io-something        89888
``

Reply:
Sure, that's fine. I'm wondering if this should be it's own command really with the default just be displaying the pid for the shim you specified and then maybe an -all flag, but that's what I've done here and it feels odd to have them split. hmm